### ISSUE TITLE
Tri Force Heroes stuck on triforce glitch

### ISSUE THREAD

**Issue by helloyanis** [PretendoNetwork/Pretendo#50]
I have found a glitch and the steps to reproduce it in Tri Force Heroes.

1. Search for teammates, with no special challenge.
2. Have a second console search for teammates, with special challenge

What should happen : 
- The two players don't get matched together

What happens : 
- The two players get matched together
- Whenever a 3rd player will join, the game locks up when players stand on the tri force because the game fails to load the level select. People can still emote but no one sees the level select, forcing a player to disconnect

Why this happens : 
- I suspect it's because filters don't get applied (for example slow and steady or fast, or with/without challenge) and people still get matched together. For "Slow and steady" or "Fast game" it shouldn't cause issues and allow to find more players which is cool, but the game doesn't know which level select (with or without challenge) it should load when people with different challenge preferences match up.

**Comment by jonbarrow** [comment]
CC @shutterbug2000

**Comment by helloyanis** [comment]
Some precisions on this, I was able to match in a working room with challenges by selecting no challenge (and the triforce worked). So I think this happens when a player has not unlocked special challenges yet (from a save file before beating the first world, the woodlands) matches in a challenge room.
Since I already 100% the game, I think that's why it worked.

**Comment by AwesomeTMC** [comment]
Same thing happens when everyone selects the same thing ("Slow and steady"), so I don't think it has to do with that. The game was working fine before today, we even got some challenges for woodlands in, but it just seems to have stopped working suddenly.

**Comment by helloyanis** [comment]
Yes, that's what I also found out. Maybe [this](https://github.com/PretendoNetwork/Pretendo/issues/50#issuecomment-2053717262) is the reasol it happens?

**Comment by AwesomeTMC** [comment]
Me and my friends all completed the same missions together (we did full resets so we could do it all together), so it wouldn't be that either. All of us have the challenges unlocked, so I'm unsure what's causing this. It does *seem* to be related to challenges though. I haven't seen it happening with anything else.
